custom_metrics/fugu-apis.js

```diff
@@ -0,0 +1,213 @@
+    where: "JavaScript",
+  },
+  "WebSerial": {
+    regEx: /navigator\.serial\.requestPort\(/g,
+    where: "JavaScript",
+  },
+  "WebNFC": {
+    regEx: /new NDEFReader\(/g,
+    where: "JavaScript",
+  },
+  "Run On Login": {
+    regEx: /navigator\.runOnOsLogin\.set\(/g,
+    where: "JavaScript",
+  },
+  "WebCodecs": {
+    regEx: /new MediaStreamTrackProcessor\(/g,
+    where: "JavaScript",
+  },
+  "Digital Goods": {
+    regEx: /getDigitalGoodsService\(/g,
+    where: "JavaScript",
+  },
+  "Idle Detection": {
+    regEx: /new IdleDetector\(/g,
+    where: "JavaScript",
+  },
+  "Storage Foundation": {
+    regEx: /storageFoundation\.open\(/g,
+    where: "JavaScript",
+  },
+  "Handwriting Recognition": {
+    regEx: /navigator\.queryHandwritingRecognizerSupport\(/g,
+    where: "JavaScript",
+  },
+  "Compute Pressure": {
+    regEx: /new ComputePressureObserver\(/g,
+    where: "JavaScript",
+  },
+  "Ambient Light Sensor": {
+    regEx: /new AmbientLightSensor\(\)/g,
+    where: "JavaScript",
+  },
+  "File Handling": {
+    regEx: /launchQueue\.setConsumer\(/g,
+    where: "JavaScript",
+  },
+  "Notification Triggers": {
+    regEx: /showTrigger\:/g,
+    where: "JavaScript",
+  },
+  "Local Font Access": {
+    regEx: /navigator\.fonts\.query\(\);/g,
+    where: "JavaScript",
+  },
+  "Multi-Screen Window Placement": {
+    regEx: /getScreens\(\)/g,
+    where: "JavaScript",
+  },
+  "WebSocketStream": {
+    regEx: /new WebSocketStream\(/g,
+    where: "JavaScript",
+  },
+  "WebTransport": {
+    regEx: /new WebTransport\(/g,
+    where: "JavaScript",
+  },
+  "Web Share Target": {
+    regEx: /"share_target"\:/g,
+    where: "Web App Manifest",
+  },
+  "Web Share Target (Files)": {
+    regEx: /"enctype"\: "multipart\/form\-data"/g,
+    where: "Web App Manifest",
+  },
+  "Shortcuts": {
+    regEx: /"shortcuts"\:/g,
+    where: "Web App Manifest",
+  },
+  "Declarative Link Capturing": {
+    regEx: /"capture_links"\:/g,
+    where: "Web App Manifest",
+  },
+  "Tabbed Application Mode": {
+    regEx: /"tabbed"/g,
+    where: "Web App Manifest",
+  },
+  "Window Controls Overlay": {
+    regEx: /"window\-controls\-overlay"/g,
+    where: "Web App Manifest",
+  },
+  "URL Handlers": {
+    regEx: /"url_handlers"/g,
+    where: "Web App Manifest",
+  },
+  "Protocol Handlers": {
+    regEx: /"protocol_handlers"/g,
+    where: "Web App Manifest",
+  },
+};
+
+// To make sure we don't match on, e.g., blog posts that contain the patterns,
+// make sure that the file names fulfill certain conditions as a heuristic.
+const checkURLConditions = (where, url) => {
+  // If the pattern has to occur in JavaScript, make sure the file name
+  // includes either `.js` or `.mjs`.
+  if (where === "JavaScript" && /\.m?js/.test(url)) {
+    return true;
+    // If the pattern has to occur in the Web App Manifest, make sure the file
+    // name includes either `.json` or `.webmanifest`.
+  } else if (
```

Comment:
Nit: Remove `else if` because `return true` above.
Nit: Use `\.webmanifest|\.json/`

```js
// If the pattern has to occur in JavaScript, make sure the file name
// includes either `.js` or `.mjs`.
if (where === "JavaScript" && /\.m?js/.test(url)) return true;

// If the pattern has to occur in the Web App Manifest, make sure the file
// name includes either `.json` or `.webmanifest`.
if (where === "Web App Manifest" && /\.webmanifest|\.json/.test(url))
  return true;

return false;

  ```

Reply:
Changed the RegEx, but I need to have a fall-through for the `return false` case.

Reply:
I don't get the "fall-through". Can you explain?

Reply:
I may not get what you're saying. The way I understand it is rather than… 

```js
if (a) {
  return true;
} else if (b) {
  return true;
}
// Fall-through
return false;
```

…you want me to write…

```js
if (a) {
  return true;
}
if (b) {
  return true;
}
// Fall-through
return false;
```

Is this what you mean?

Reply:
Yes. This pattern is used a lot in Chromium source code base. It simplifies reading a lot.